### DISCUSSION TITLE
Routing issue.

### DISCUSSION THREAD

**Discussion by jwstemper** [Dotnet-Boxed/Templates#116] [Q&A]
Having troubles getting @Url.RouteUrl to show proper route. It works for many of my controllers and actions but this one is a mystery to me.

```
 public static class ProofingControllerAction
    {
        public const string Index = "Index";
        public const string proofCampaign = "proofCampaign";
        public const string proofOrder = "proofOrder";
    }
    public static class ProofingControllerRoute
    {
        public const string ProofCampaign = ControllerName.Proof + "ProofCampaign";
        public const string ProofOrder = ControllerName.Proof + "ProofOrder";
        public const string GetCampaigns = ControllerName.Proof + "GetCampaigns";

    }
```

My Action:

```
 [Route("proofOrder/{OrderId}", Name = ProofingControllerRoute.ProofOrder)]
        public ActionResult ProofOrder(int orderId)
        {
            using (var db = new ProofingContext())
            {
                var order = db.Orders.Find(orderId);
                return View(order);
            } }
```

The view that the I am trying to insert the @Url.RouteUrl into is a different view then the one that goes with this action. I can print out the name via  @ProofingControllerRoute.ProofOrder and it shows on the page but the RouteUrl comes up blank.

**Comment by jwstemper** [comment]
Figured it out. Changed:
`[Route("proofOrder/{OrderId}", Name = ProofingControllerRoute.ProofOrder)]`
to
`[Route("proofOrder/{OrderId?}", Name = ProofingControllerRoute.ProofOrder)]`

Not sure why the "?" makes a difference. Still getting used to Attribute routing.

**Comment by RehanSaeed** [comment]
`?` makes the parameter optional. In the method you show above `orderId` is a `int` and not a `int?` so you should not need to use `?` in the route.

Keep the case of your parameters in your route the same as the parameter in the method (`OrderId` vs `orderId`).

Apart from the casing issue (not sure if this is an issue, I always keep the case the same and don't know what happens when you don't), I don't see from your code sample why you are experiencing that behavior.
